src/ATen/native/xpu/sycl/DilatedMaxPool3d.cpp

```diff
@@ -0,0 +1,779 @@
+      int itime,
+      int iheight,
+      int iwidth,
+      int obatch,
+      int otime,
+      int oheight,
+      int owidth,
+      int kT,
+      int kH,
+      int kW,
+      int dT,
+      int dH,
+      int dW,
+      int pT,
+      int pH,
+      int pW,
+      int dilationT,
+      int dilationH,
+      int dilationW,
+      int offsetZ,
+      bool channels_last)
+      : inputData_(inputData),
+        outputData_(outputData),
+        indicesData_(indicesData),
+        features_(features),
+        itime_(itime),
+        iheight_(iheight),
+        iwidth_(iwidth),
+        obatch_(obatch),
+        otime_(otime),
+        oheight_(oheight),
+        owidth_(owidth),
+        kT_(kT),
+        kH_(kH),
+        kW_(kW),
+        dT_(dT),
+        dH_(dH),
+        dW_(dW),
+        pT_(pT),
+        pH_(pH),
+        pW_(pW),
+        dilationT_(dilationT),
+        dilationH_(dilationH),
+        dilationW_(dilationW),
+        offsetZ_(offsetZ),
+        channels_last_(channels_last) {}
+
+ private:
+  const scalar_t* inputData_;
+  scalar_t* outputData_;
+  int64_t* indicesData_;
+  int features_;
+  int itime_;
+  int iheight_;
+  int iwidth_;
+  int obatch_;
+  int otime_;
+  int oheight_;
+  int owidth_;
+  int kT_;
+  int kH_;
+  int kW_;
+  int dT_;
+  int dH_;
+  int dW_;
+  int pT_;
+  int pH_;
+  int pW_;
+  int dilationT_;
+  int dilationH_;
+  int dilationW_;
+  int offsetZ_;
+  bool channels_last_;
+};
+
+template <typename scalar_t>
+static void max_pool3d_with_indices_out_frame(
+    const scalar_t* inputData,
+    const Tensor& output,
+    const Tensor& indices,
+    int features,
+    int64_t totalZ,
+    int itime,
+    int iheight,
+    int iwidth,
+    int obatch,
+    int otime,
+    int oheight,
+    int owidth,
+    int kT,
+    int kH,
+    int kW,
+    int dT,
+    int dH,
+    int dW,
+    int pT,
+    int pH,
+    int pW,
+    int dilationT,
+    int dilationH,
+    int dilationW,
+    bool channels_last) {
+  int offsetZ = 0;
+  while (totalZ > 0) {
+    MaxPool3dWithIndicesOutFrameImplKernelFunctor<scalar_t> kfn(
+        inputData,
+        output.mutable_data_ptr<scalar_t>(),
+        indices.mutable_data_ptr<int64_t>(),
+        features,
+        itime,
+        iheight,
+        iwidth,
+        obatch,
+        otime,
+        oheight,
+        owidth,
+        kT,
+        kH,
+        kW,
+        dT,
+        dH,
+        dW,
+        pT,
+        pH,
+        pW,
+        dilationT,
+        dilationH,
+        dilationW,
+        offsetZ,
+        channels_last);
+
+    // width size is fixed size = 32, height dim equals = syclMaxWorkGroupSize
+    // / width_size
+    int width_group_size = 32;
+    int height_group_size = syclMaxWorkGroupSize(kfn) / width_group_size;
+    int width_group_range = ceil_div<int>(owidth, width_group_size);
+    int height_group_range = ceil_div<int>(oheight, height_group_size);
+
+    int z_group_range = totalZ > 65535 ? 65535 : totalZ;
```

Comment:
65535 is CUDA's restriction for grid size. SYCL doesn't need.
so we can 1, remove this loop.
2, use max work item per tile as global range per loop.